util/argo/audit_logger.go

```diff
@@ -74,22 +83,48 @@ func (l *AuditLogger) logEvent(objMeta metav1.ObjectMeta, gvk schema.GroupVersio
 		Reason:         info.Reason,
 	}
 	logCtx.Info(message)
-	_, err := l.kIf.CoreV1().Events(l.ns).Create(&event)
+	_, err := l.kIf.CoreV1().Events(objMeta.Namespace).Create(&event)
 	if err != nil {
 		logCtx.Errorf("Unable to create audit event: %v", err)
 		return
 	}
 }
 
 func (l *AuditLogger) LogAppEvent(app *v1alpha1.Application, info EventInfo, message string) {
-	l.logEvent(app.ObjectMeta, v1alpha1.ApplicationSchemaGroupVersionKind, info, message, map[string]string{
+	objectMeta := ObjectRef{
```

Comment:
Both `v1alpha1.Application` and `v1alpha1.AppProject` come with an `metav1.ObjectMeta` object, however `ResourceNode` does not. To be able to use the `l.logEvent` function across all three types I created the function specific `ObjectMeta` object. I considered simply using the `metav1.ObjectMeta` object and filling it in, however since the `metav1.ObjectMeta` is owned by K8s and it had many extraneous fields I thought it better to create a local object that was more explicit.